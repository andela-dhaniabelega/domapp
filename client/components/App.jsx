import React from 'react';
import Header from './Header';

// import {connect} from 'react-redux';

// const mapStateToProps = state => ({appName: state.appName});

class App extends React.Component {
  render() {
    return (
      <div >
        <Header /> 
        {this.props.children}
      </div>
    );
  }
}
export default App;

// export default connect(mapStateToProps, () => ({}))(App);

