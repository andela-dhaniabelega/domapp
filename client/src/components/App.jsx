import React from 'react';
import PropTypes from 'prop-types';
import Header from './Header';

/**
 * @description Container for all components
 * @return {void}
 */
const App = () => ({
	render() {
		return (
			<div className="main">
				<Header />
				{this.props.children}
				<div className="clear" />
			</div>
		);
	}
});

App.propTypes = {
	children: PropTypes.element.isRequired
};
export default App;

