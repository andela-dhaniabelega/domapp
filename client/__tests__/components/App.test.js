import React from 'react';
import { shallow } from 'enzyme';
import App from '../../src/components/App';

describe('Footer', () => {
	it('renders a snapshot', () => {
		const wrapper = shallow(<App />);
		expect(wrapper).toMatchSnapshot();
	});
});
