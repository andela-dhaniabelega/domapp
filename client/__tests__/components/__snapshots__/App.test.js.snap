// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer renders a snapshot 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    className="main"
>
    <withRouter(Connect(Header)) />
    <div
        className="clear"
    />
</div>,
  "nodes": Array [
    <div
      className="main"
>
      <withRouter(Connect(Header)) />
      <div
            className="clear"
      />
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <App />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": Object {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {},
        "refs": Object {},
        "render": [Function],
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div
          className="main"
>
          <withRouter(Connect(Header)) />
          <div
                    className="clear"
          />
</div>,
        "_debugID": 2,
        "_renderedOutput": <div
          className="main"
>
          <withRouter(Connect(Header)) />
          <div
                    className="clear"
          />
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <App />,
}
`;
